pkg/apis/serving/v1beta1/configmap.go

```diff
@@ -114,11 +114,12 @@ type InferenceServicesConfig struct {
 
 // +kubebuilder:object:generate=false
 type IngressConfig struct {
-	IngressGateway          string `json:"ingressGateway,omitempty"`
-	IngressServiceName      string `json:"ingressService,omitempty"`
-	LocalGateway            string `json:"localGateway,omitempty"`
-	LocalGatewayServiceName string `json:"localGatewayService,omitempty"`
-	IngressDomain           string `json:"ingressDomain,omitempty"`
+	IngressGateway          string  `json:"ingressGateway,omitempty"`
+	IngressServiceName      string  `json:"ingressService,omitempty"`
+	LocalGateway            string  `json:"localGateway,omitempty"`
+	LocalGatewayServiceName string  `json:"localGatewayService,omitempty"`
+	IngressDomain           string  `json:"ingressDomain,omitempty"`
+	IngressClassName        *string `json:"ingressClassName,omitempty"`
```

Comment:
We can default the `IngressClassName` to `istio` and add a comment that only for `RawDeployment` we allow other ingress option currently.

Let's set the default to the [config](https://github.com/kserve/kserve/blob/master/config/configmap/inferenceservice.yaml#L161)  and the [test config](https://github.com/kserve/kserve/blob/master/config/overlays/test/configmap/inferenceservice.yaml#L161), add a test case without using the [ingress class annotation](https://github.com/kserve/kserve/blob/master/test/e2e/predictor/test_raw_deployment.py#L38).

Reply:
I updated `istio` as default ingress class name. We'll need to add the istio `ingressClass` creation (https://istio.io/latest/docs/tasks/traffic-management/ingress/kubernetes-ingress/#specifying-ingressclass) in the documentation since it's not created by default.

Reply:
Updated the test too